src/components/icons/LottieAnimation.tsx

```diff
@@ -1,36 +1,125 @@
+    // Check for animated color property
+    if (item.c.a === 1) {
+      item.c.k.forEach((keyframe: any) => {
+        keyframe.s = rgbaArray
+      })
+    } else {
+      item.c.k = rgbaArray
+    }
+  }
+  if (item.it) {
+    item.it.forEach((subItem: any) => updateColorsInItem(subItem, rgbaArray))
+  }
+}
+
+/**
+ * Changes the color of stroke and fill attributes in the Lottie animation data.
+ *
+ * This function traverses the Lottie animation data structure to locate all items that
+ * represent shape colors (strokes and fills) and applies a new color based on the provided
+ * RGBA values derived from the 'newColor' hexadecimal string.
+ *
+ * @param data - The Lottie animation data object containing multiple layers and shapes.
+ * Each layer is iterated through to apply the color change to eligible elements.
+ * @param newColor - A string representing the new color to be applied, in hexadecimal format.
+ * This string is converted into an RGBA array to update the animation colors.
+ */
+const changeLineColor = (data: any, newColor: string): void => {
+  const rgbaArray = hexToRGBA(newColor)
+
+  data.layers.forEach((layer: any) => {
+    if (layer.shapes) {
+      layer.shapes.forEach((shape: any) => {
+        shape.it.forEach((item: any) => updateColorsInItem(item, rgbaArray))
+      })
+    }
+  })
+}
+
 /**
  * LottieAnimation Component.
  *
- * This component renders a Lottie animation using the `lottie-react` Player.
- * It accepts animation data and an optional speed prop to control the animation speed.
+ * This component utilizes the 'lottie-react' Player to render a Lottie animation.
+ * It accepts animation data and supports customizable speed and color properties,
+ * allowing for dynamic control over the animation's playback rate and the stroke
+ * and fill colors of its elements.
  *
+ * Props:
+ * animationData: The JSON data that defines the Lottie animation.
+ * Speed: Optional. A number that specifies the playback speed of the animation. Defaults to 1 for normal speed.
+ * Color: Optional. A string representing the hex color code used to update the stroke and fill colors of the animation. Defaults to '#FFFFFF' (white) if not specified.
+ * OnComplete: Optional. A callback function that is triggered when the animation completes its playback.
+ * Style: Optional. A React.CSSProperties object to apply inline styles to the player container.
+ * This does not affect the Lottie animation's JSON content or alter its visual elements.
  */
 const LottieAnimation: React.FC<LottieAnimationProps> = ({
   animationData,
-  speed = 1, // Default speed set to 1x
+  speed = 1,
+  color = '#FFFFFF',
   onComplete,
   style,
 }) => {
+  const [key, setKey] = useState(0)
   const lottieRef = useRef<LottieRefCurrentProps | null>(null)
 
   useEffect(() => {
     if (lottieRef.current) {
       lottieRef.current.setSpeed(speed)
+      if (animationData && color) {
+        changeLineColor(animationData, color)
```

Comment:
`changeLineColor` mutates the inside of `animationData` without the object reference changing. This results in the awkward external `key` required notify the component that the value has actually changed.

Instead, use `useMemo` to create a derived `animationDataWithColor` that can be passed to `Player`. Then the Player component will update naturally when the `animationDataWithColor` reference changes. It's easier to reason about because it avoids mutations, and it's friendlier to React since it follows norms around object references.